7.2.2 broke some Array types when used in typeguards

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

7.2.2

### Node.js version

20.2.0

### MongoDB server version

unrelated

### Typescript version (if applicable)

4.9.5

### Description

mongoose 7.2.2 (via #13446) broke some types that are used in typegoose, but only typeguards that use `mongoose.Types.Array`

fixes would be to either remove the following line, or remove the `<T>` from the line
https://github.com/Automattic/mongoose/blob/6b820a71fd7f127cc7c0cdfc99b3ca74dd8b92c2/types/types.d.ts#L45

both resolve the issue, i dont understand why typescript complains, but the mentioned solution(s) above resolve it for this case, but i dont know if this would break some types tests

full error (from the reproduce script):

```txt
src/test.ts:11:12 - error TS2677: A type predicate's type must be assignable to its parameter's type.
  Type 'Array<S>' is not assignable to type 'Array<PopulatedDoc<T, S>>'.
    Types of property 'toObject' are incompatible.
      Type '{ (options?: ToObjectOptions<S> | undefined): any; <T>(options?: ToObjectOptions<T> | undefined): T; }' is not assignable to type '{ (options?: ToObjectOptions<PopulatedDoc<T, S>> | undefined): any; <T>(options?: ToObjectOptions<T> | undefined): T; }'.
        Types of parameters 'options' and 'options' are incompatible.
          Type 'ToObjectOptions<PopulatedDoc<T, S>> | undefined' is not assignable to type 'ToObjectOptions<S> | undefined'.
            Type 'ToObjectOptions<PopulatedDoc<T, S>>' is not assignable to type 'ToObjectOptions<S>'.
              Types of property 'transform' are incompatible.
                Type 'boolean | ((doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any) | undefined' is not assignable to type 'boolean | ((doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any) | undefined'.
                  Type '(doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any' is not assignable to type 'boolean | ((doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any) | undefined'.
                    Type '(doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any' is not assignable to type '(doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any'.
                      Types of parameters 'options' and 'options' are incompatible.
                        Type 'ToObjectOptions<S>' is not assignable to type 'ToObjectOptions<PopulatedDoc<T, S>>'.
                          Types of property 'transform' are incompatible.
                            Type 'boolean | ((doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any) | undefined' is not assignable to type 'boolean | ((doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any) | undefined'.
                              Type '(doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any' is not assignable to type 'boolean | ((doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any) | undefined'.
                                Type '(doc: S, ret: Record<string, any>, options: ToObjectOptions<S>) => any' is not assignable to type '(doc: PopulatedDoc<T, S>, ret: Record<string, any>, options: ToObjectOptions<PopulatedDoc<T, S>>) => any'.
                                  Types of parameters 'doc' and 'doc' are incompatible.
                                    Type 'PopulatedDoc<T, S>' is not assignable to type 'S'.
                                      'S' could be instantiated with an arbitrary type which could be unrelated to 'PopulatedDoc<T, S>'.

11 ): docs is mongoose.Types.Array<S> {
              ~~~~~~~~~~~~~~~~~~~~~~~

  src/test.ts:8:32
    8 export function isRefTypeArray<T, S extends mongoose.RefType>(
                                     ~
    This type parameter might need an `extends S` constraint.
```
(the typescript suggestion of `This type parameter might need an "extends S" constraint.` is not a solution)

### Steps to Reproduce

```ts
/* eslint-disable @typescript-eslint/ban-types */
// NodeJS: 19.9.0
// MongoDB: 5.0 (Docker)
// Typescript 4.9.5
import * as mongoose from 'mongoose'; // mongoose@7.2.2

export function isRefTypeArray<T, S extends mongoose.RefType>(
  docs: mongoose.Types.Array<mongoose.PopulatedDoc<T, S>> | null | undefined
  // error on the next line
): docs is mongoose.Types.Array<S> {
  return Array.isArray(docs) && docs.every((v) => /* some testing func */ true);
}
```

Reproduction Repository & Branch: https://github.com/typegoose/typegoose-testing/tree/mongooseToObjectOptions

### Expected Behavior

no typescript error

--- a/types/types.d.ts
+++ b/types/types.d.ts
@@ -42,7 +42,7 @@ declare module 'mongoose' {
       shift(): T;
 
       /** Returns a native js Array. */
-      toObject(options?: ToObjectOptions<T>): any;
+      toObject(options?: ToObjectOptions): any;
       toObject<T>(options?: ToObjectOptions<T>): T;
 
       /** Wraps [`Array#unshift`](https://developer.mozilla.org/en/JavaScript/Reference/Global_Objects/Array/unshift) with proper change tracking. */

